odoo/addons/base/models/res_currency.py

```diff
@@ -52,6 +52,23 @@ class Currency(models.Model):
         ('rounding_gt_zero', 'CHECK (rounding>0)', 'The rounding factor must be greater than 0!')
     ]
 
+    def write(self, vals):
+        res = super().write(vals)
+        if 'active' not in vals:
+            return res
```

Comment:
What about currencies creation or deletion ?
You should cover those cases too.

Reply:
added the `create()` and `unlink()` methods